### PR TITLE
All sources endpoint

### PR DESCRIPTION
**Pull request by hitarthpattani**
### Description (*)

Create an endpoint which provides source records based on criteria provided with API call. 

### Fixed Issues (if relevant)

1. magento/async-import#86: Get All sources API endpoint

### Manual testing scenarios (*)

URL (GET request): [DOMAIN]/rest/V1/import/sources?criteria
Headers: Authorization: Bearer <admin_user_token>

**Expected Result:**

```
{
    "items": [
        {
            "entity_id": "1",
            "uuid": "7ed24a77-2d72-432d-905c-6d67e4bccf37",
            "status": "uploaded",
            "source_type": "csv",
            "import_type": "local_path",
            "import_data": "7ed24a77-2d72-432d-905c-6d67e4bccf37.csv",
            "created_at": "2019-04-11 11:08:51",
            "format": {
                "csv_separator": ",",
                "csv_enclosure": "\"",
                "csv_delimiter": ",",
                "multiple_value_separator": "|",
                "mapping": [
                    {
                        "source_attribute": "string",
                        "destination_attribute": "string",
                        "processing_rules": "string",
                        "taxonomy": "string",
                        "values_mapping": [
                            {
                                "old_value": "value1",
                                "new_value": "value2"
                            },
                            {
                                "old_value": "value3",
                                "new_value": "value4"
                            }
                        ]
                    }
                ]
            }
        },
        {
            "entity_id": "2",
            "uuid": "c8598392-e0d7-4a3f-b0b3-ae202b3d6c7e",
            "status": "uploaded",
            "source_type": "csv",
            "import_type": "local_path",
            "import_data": "c8598392-e0d7-4a3f-b0b3-ae202b3d6c7e.csv",
            "created_at": "2019-04-11 11:09:41",
            "format": {
                "csv_separator": ",",
                "csv_enclosure": "\"",
                "csv_delimiter": ",",
                "multiple_value_separator": "|",
                "mapping": [
                    {
                        "source_attribute": "string",
                        "destination_attribute": "string",
                        "processing_rules": "string",
                        "taxonomy": "string",
                        "values_mapping": [
                            {
                                "old_value": "value1",
                                "new_value": "value2"
                            },
                            {
                                "old_value": "value3",
                                "new_value": "value4"
                            }
                        ]
                    }
                ]
            }
        }
    ],
    "search_criteria": {
        "filter_groups": []
    },
    "total_count": 2
}
```

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by nuzil** [comment]
Also I'm getting next error if there are no mapping set: 

```
"Fatal Error: 'Uncaught Error: Call to a member function getMapping() on null in \magento\/async-import\/app\/code\/Magento\/ImportService\/Model\/SourceRepository.php:150\nStack trace:\n#0 
```

**Comment by nuzil** [comment]
Lets implement Delete operation by using Technical Guidelines
https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html
6.4.2.3

Good example:
https://github.com/magento-engcom/msi/blob/2.3-develop/app/code/Magento/Inventory/Model/StockRepository.php

**Comment by hitarthpattani** [comment]
Hello @nuzil 

I have changed the "criteria" param to "searchCriteria", resolve getMapping() on null issue and implement getList operation by using Technical Guidelines.

Please check and let me know if there is any further change.

**Comment by hitarthpattani** [comment]
Hello @nuzil I have resolved the conflicts to the branch and change the base branch to asynchronous-import-module as discussed. Please let me know if there is any change needed to merge this branch.

File: app/code/Magento/ImportService/Model/SourceRepository.php [hunk 1]
```diff
@@ -10,6 +10,7 @@
 use Magento\Framework\Api\SearchCriteriaInterface;
 use Magento\Framework\Api\SearchResultsInterfaceFactory;
 use Magento\Framework\Api\SearchResultsInterface;
+use Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface;
 use Magento\Framework\Api\SortOrder;
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/SourceRepository.php]: Is it ever used? If not please remove

File: app/code/Magento/ImportService/Model/SourceRepository.php [hunk 2]
```diff
@@ -10,6 +10,7 @@
 use Magento\Framework\Api\SearchCriteriaInterface;
 use Magento\Framework\Api\SearchResultsInterfaceFactory;
 use Magento\Framework\Api\SearchResultsInterface;
+use Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface;
```
Comment (nuzil) [inline app/code/Magento/ImportService/Model/SourceRepository.php]: Is it ever used? If not please remove

File: app/code/Magento/ImportService/Model/SourceRepository.php (hunk 1 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/SourceRepository.php]: Hello @nuzil I have removed all extra dependencies.

File: app/code/Magento/ImportService/Model/SourceRepository.php (hunk 2 above)
Comment (hitarthpattani) [reply app/code/Magento/ImportService/Model/SourceRepository.php]: Hello @nuzil I have removed all extra dependencies.

**Pull request merged**